File: Problem #1/problem1_solution.py
import numpy as np


def doesArrayContainK(arr, k):
    """Return the two numbers that add up to k
    Parameters:
    argument1 (array): An array of numbers
    argument2 (int): The number k

    Returns:
    Tupple: The two numbers that equals k
    """

    current_val = 0
    for i in arr:
        for j in arr:
            if (i != j):
                current_val = i + j
                if (current_val == k):
                    print("True: " + str(i) + " + " + str(j) + " = " + str(k))
                    return

    if (current_val != k):
        print("False")

###Harcoded example solution 
example_array = [10, 15, 3, 7]

k = 17

doesArrayContainK(example_array, k)

###Random generating numbers
#Create an array with size and containing numbers ranging from lowest to highets
random_array = np.random.randint(low = 1, high = 100, size = 4)

#Find highest possible sum of two numbers in the random array
max = 0
for i in random_array:
    for j in random_array:
        if (i != j):
            current_highest = i + j
            if (max < current_highest):
                max = current_highest

random_k = np.random.randint(low = 1, high = max)

doesArrayContainK(random_array, random_k)

File: Problem #1/problem1_solution.js
/**
 * See if two numbers in the list add up to the value k
 * 
 * @param {List} arr List of numbers
 * @param {Number} k The value we are looking for
 * 
 * @return {Boolean} True if k exists in the list, false otherwise
 */
function doesArrayAddUpToK(arr, k) {
    var current_val = 0
    for (i in arr) {
        for (j in arr) {
            if (arr[i] !== arr[j]) {
                current_val = arr[i] + arr[j]
                // console.log(current_val)
                //Triple equals for matching on value and type
                if (current_val === k) { 
                    return true
                }
            }    
        }
    }
    return false
}

var example_array =  [10, 15, 3, 7]
var k = 17

console.log(doesArrayAddUpToK(example_array, k))
